src/simd/x86.rs

```diff
@@ -0,0 +1,68 @@
+use crate::simd::SimdValueOps;
+
+#[cfg(target_arch = "x86")]
+use core::arch::x86 as arch;
+#[cfg(target_arch = "x86_64")]
+use core::arch::x86_64 as arch;
+use core::mem;
+
+#[derive(Copy, Clone)]
+pub struct SimdValue(arch::__m128i);
+
+impl SimdValueOps for SimdValue {
+    #[inline]
+    #[target_feature(enable = "sse2")]
+    unsafe fn xor(self, value: u64) -> Self {
```

Comment:
These seem like safe interfaces. What are the invariants that need to be held to use these functions safely?

Reply:
They are safe if the target supports them. Intrinsics are forced to be emitted even if the target doesn't support them, therefore its required to either check at run or compile-time if they are actually supported.

Reply:
If `SimdValue` is gated at compile-time, then I think a safe abstraction layer might be useful. Somewhat similar to https://docs.rs/safe_arch/latest/safe_arch/struct.m128i.html.